cupyx/scipy/signal/_filter_design.py

```diff
@@ -0,0 +1,332 @@
+import operator
+from math import pi
+
+import cupy
+from cupy.polynomial.polynomial import (polyval as npp_polyval,
+        polyvalfromroots as npp_polyvalfromroots)
+import cupyx.scipy.fft as sp_fft
+
+
+def _try_convert_to_int(x):
+    """Return an integer for ``5`` and ``array(5)``, fail if not an
+    integer scalar.
+
+    NB: would be easier if ``operator.index(cupy.array(5))`` worked
+    (numpy.array(5) does)
+    """
+    if isinstance(x, cupy.ndarray):
+        if x.ndim == 0:
+            value = x.item()
+        else:
+            return x, False
```

Comment:
`x.item()` makes device-host synchronization, so I think returning cupy.ndarray is okay.

Reply:
Sadly, a python integer is required down the line: 

```
(Pdb) start, stop, num
(0, 3.141592653589793, array(7))
(Pdb) cupy.linspace(start, stop, num)
*** TypeError: 'ndarray' object cannot be interpreted as an integer
```

Reply:
How about also using the following branch for 0-dim cupy.ndarray?

https://github.com/cupy/cupy/blob/483c169ed597e6a2468738b0893b8a1732aa6125/cupyx/scipy/signal/_filter_design.py#L168-L169

Reply:
That would change the interface: for `worN=asarray(5)` currently means that the frequencies are `w = linspace(0, 2*pi, 5)`, and that change would make it `w = np.asarray([5])`.

We can probably just disallow 0D arrays and make worN argument to be either a python int or a 1D array, would that work?

Reply:
It's fine, thanks for the explanation!